services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy

```diff
@@ -0,0 +1,134 @@
+package org.openkilda.functionaltests.spec.stats
+
+
+import static org.openkilda.functionaltests.extension.tags.Tag.VIRTUAL
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.traffexam.TraffExamService
+import org.openkilda.testing.tools.FlowTrafficExamBuilder
+
+import groovy.time.TimeCategory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+import spock.lang.Shared
+
+import javax.inject.Provider
+
+@Narrative("""Now we have two FL instances: Management and Statistics.
+- FL Stats: collect statistics only from the switches.
+- FL Management: do the other work and can collect statistics as well when a switch doesn't connect to FL Stats.""")
+class MflStatSpec extends BaseSpecification {
+    @Shared
+    @Value('${opentsdb.metric.prefix}')
+    String metricPrefix
+
+    @Value('${floodlight.controller.management}')
+    private String managementController
+
+    @Value('${floodlight.controller.stat}')
+    private String statController
+
+    @Autowired
+    Provider<TraffExamService> traffExamProvider
+
+    @Tags(VIRTUAL)
+    def "System is able to collect stats from the statistic and management controllers"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected
+        // not all OVS switches are connected to both controllers by default
+        lockKeeper.setController(srcSwitch.dpId, managementController + " " + statController)
```

Comment:
This brings constant changes to the topology state, i.e. switch that was originally connected to only 1 controller will be connected to 2 controllers after this test and remain such for all subsequent tests. This means that result of subsequent tests now depend on the order of when they happen - before or after _this_ test. This is unacceptable.
First option would be to update TopologyDefinition.Switch with additional property(-ies) that will allow to specify whether certain switch should be connected to Stats controller or not. This property will also be respected by `reviveSwitch` method, which will now have to look at this new property and add controller(s) respectively.

Another option will be to step back and leave all switches connected to both controllers by default (as it was previously).

Reply:
return as it was before
(now both controllers will be set on all switches)